docs/developer-docs/latest/development/typescript.md

```diff
@@ -132,3 +128,84 @@ const appContext = await strapi.compile();
 const app = await strapi(appContext);
 
 ```
+
+## Add TypeScript support to an existing Strapi project
+
+Adding TypeScript support to an existing project requires modifications to the project files. Additionally the TypeScript flag `allowJs` should be set to `true` in the root `tsconfig.json` file to incrementally add TypeScript files to existing JavaScript projects. The `allowJs` flag allows `.ts` and `.tsx` files to coexist with JavaScript files. TypeScript support can be added to an existing Strapi project using the following procedure:
```

Comment:
> Adding TypeScript support to an existing project requires modifications to the project files. 

I guess you mean that devs will need to (gradually) convert each JS file to a TS file and take advantage of TS features and syntax, right? Maybe we could be a bit more explicit, then, what do you think?

And this is also why I think the "conversion" to TS is not so "short" 😅 (see my other comment above). Adding the `tsconfig` files is fast, but converting all the files… not so much 😅 So I guess it really depends on the wording we chose.

Reply:
I see what you mean. I added a bit more detail to make it clear.